tests/python/TestEntities.py

```diff
@@ -89,33 +94,69 @@ def findEntity(self, thID):
 
         response, res_arr = db.query(all_queries)
 
-        self.assertEqual(response[0]["FindEntity"]["status"], 0)
-        self.assertEqual(response[0]["FindEntity"]["entities"][0]
-                                    ["lastname"], "Ferro")
-        self.assertEqual(response[0]["FindEntity"]["entities"][0]
-                                    ["threadid"], thID)
+        try:
+
+            self.assertEqual(response[0]["FindEntity"]["status"], 0)
+            self.assertEqual(response[0]["FindEntity"]["entities"][0]
+                                        ["lastname"], "Ferro")
+            self.assertEqual(response[0]["FindEntity"]["entities"][0]
+                                        ["threadid"], thID)
+        except:
+            results[thID] = -1
+
+        results[thID] = 0
+
+    def test_runMultipleAdds(self):
+
+        # Test concurrent AddEntities
+        concurrency = 32
+        thread_arr = []
+        results = [None] * concurrency
+        for i in range(0,concurrency):
+            thread_add = Thread(target=self.addEntity,args=(i, results) )
+            thread_add.start()
+            thread_arr.append(thread_add)
```

Comment:
What are these set of changes for? I am lazy to figure it out myself :) seems like a different commit from a quick scan.

Reply:
stuff for checking that concurrent read/writes worked. you only need to worry about it if the test fails :)